includes/data-stores/class-wc-customer-download-log-data-store.php

```diff
@@ -0,0 +1,196 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC Customer Download Log Data Store.
+ *
+ * @version  3.3.0
+ * @category Class
+ * @author   WooThemes
+ */
+class WC_Customer_Download_Log_Data_Store implements WC_Customer_Download_Log_Data_Store_Interface {
+
+	/**
+	 * Create download log entry.
+	 *
+	 * @param WC_Customer_Download_Log $download_log
+	 */
+	public function create( &$download_log ) {
+		global $wpdb;
+
+		// Always set a timestamp
+		if ( is_null( $download_log->get_timestamp( 'edit' ) ) ) {
+			$download_log->set_timestamp( current_time( 'timestamp', true ) );
+		}
+
+		$data = array(
+			'timestamp'           => date( 'Y-m-d H:i:s', $download_log->get_timestamp( 'edit' )->getTimestamp() ),
+			'permission_id'       => $download_log->get_permission_id( 'edit' ),
+			'user_id'             => $download_log->get_user_id( 'edit' ),
+			'user_ip_address'     => $download_log->get_user_ip_address( 'edit' ),
+		);
+
+		$format = array(
+			'%s',
+			'%s',
+			'%s',
+			'%s',
+		);
+
+		$result = $wpdb->insert(
+			$wpdb->prefix . 'woocommerce_downloadable_product_download_log',
+			apply_filters( 'woocommerce_downloadable_product_download_log_data', $data ),
+			apply_filters( 'woocommerce_downloadable_product_download_log_format', $format, $data )
+		);
+
+		do_action( 'woocommerce_downloadable_product_download_log', $data );
+
+		if ( $result ) {
```

Comment:
That's a good point - I'll add a `wp_die` with relevant messaging just in case something fails during the insert.  That should stop the download endpoint to make sure downloads don't continue until they're being tracked properly.